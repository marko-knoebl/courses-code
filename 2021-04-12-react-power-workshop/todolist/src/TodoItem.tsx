type TodoItemProps = {
  title: string;
  completed: boolean;
  onCompletedChange: (completed: boolean) => void;
  onDelete: () => void;
};

function TodoItem(props: TodoItemProps) {
  return (
    <li>
      <input
        type="checkbox"
        checked={props.completed}
        onChange={(event) => {
          props.onCompletedChange(event.target.checked);
        }}
      />
      {props.completed ? "DONE: " : "TODO: "}
      {props.title}
      <button onClick={() => props.onDelete()}>delete</button>
    </li>
  );
}

export default TodoItem;
