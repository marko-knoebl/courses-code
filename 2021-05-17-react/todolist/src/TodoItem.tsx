/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import { Button } from "@material-ui/core";
import { useContext } from "react";
import ThemeContext from "./ThemeContext";

type TodoItemProps = {
  title: string;
  completed: boolean;
  onToggle: () => void;
  onDelete: () => void;
};

function TodoItem(props: TodoItemProps) {
  const theme = useContext(ThemeContext);

  return (
    <li
      css={css`
        padding: 4px;
        margin: 4px;
        text-decoration: ${props.completed ? "line-through" : "none"};
        background-color: ${theme === "dark" ? "darkblue" : "lightblue"};
      `}
    >
      <input
        type="checkbox"
        checked={props.completed}
        onChange={() => {
          props.onToggle();
        }}
      />
      {props.completed ? "done: " : "todo: "}
      {props.title}
      <Button onClick={() => props.onDelete()}>delete</Button>
      <Button onClick={() => props.onToggle()}>toggle</Button>
    </li>
  );
}

export default TodoItem;
