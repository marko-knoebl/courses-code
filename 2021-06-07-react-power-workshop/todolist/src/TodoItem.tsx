import { Button } from "@material-ui/core";
import { useContext } from "react";
import ThemeContext from "./ThemeContext";

import styles from "./TodoItem.module.scss";
import classNames from "classnames";

type TodoItemProps = {
  title: string;
  isCompleted: boolean;
  onDelete: () => void;
  onCompletedChange: (completed: boolean) => void;
};

function TodoItem(props: TodoItemProps) {
  const themeContext = useContext(ThemeContext);
  return (
    <li
      style={{
        padding: 8,
        backgroundColor: themeContext === "default" ? "white" : "black",
        color: themeContext === "default" ? "black" : "white",
      }}
      className={classNames({
        [styles.todo]: true,
        [styles.completed]: props.isCompleted,
      })}
    >
      <input
        type="checkbox"
        checked={props.isCompleted}
        onChange={(event) => props.onCompletedChange(event.target.checked)}
      />
      {props.title}
      <Button variant="contained" onClick={() => props.onDelete()}>
        X
      </Button>
    </li>
  );
}

export default TodoItem;
