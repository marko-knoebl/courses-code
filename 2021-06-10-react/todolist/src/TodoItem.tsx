import { useContext } from "react";
import ThemeContext, { useThemeContext } from "./ThemeContext";
import { Todo } from "./types";

type TodoItemProps = {
  todo: Todo;
  onCompletedChanged: (completed: boolean) => void;
  onDelete: () => void;
};

function TodoItem(props: TodoItemProps) {
  const theme = useThemeContext();
  return (
    <li style={{ backgroundColor: theme === "default" ? "white" : "grey" }}>
      <input
        type="checkbox"
        checked={props.todo.completed}
        onChange={(event) => {
          props.onCompletedChanged(event.target.checked);
        }}
      />
      <span
        style={{
          color: props.todo.completed ? "grey" : "black",
          textDecoration: props.todo.completed ? "line-through" : "none",
        }}
      >
        {props.todo.completed ? "DONE: " : "TODO: "}
        {props.todo.title}
      </span>
      <button onClick={() => props.onDelete()}>delete</button>
    </li>
  );
}

export default TodoItem;
