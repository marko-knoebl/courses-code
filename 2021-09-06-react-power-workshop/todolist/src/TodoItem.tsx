type TodoItemProps = {
  id: number;
  title: string;
  completed: boolean;
  onCompletedChange: (completed: boolean) => void;
  onDelete: () => void;
};

const todoItemBaseStyle = {
  margin: 8,
  padding: 8,
};

function TodoItem(props: TodoItemProps) {
  return (
    <li
      key={props.id}
      style={{
        ...todoItemBaseStyle,
        backgroundColor: props.completed ? "lightgrey" : "lightblue",
      }}
    >
      <input
        type="checkbox"
        checked={props.completed}
        onChange={(event) => props.onCompletedChange(event.target.checked)}
      />
      {props.title}
      <button onClick={() => props.onDelete()}>delete</button>
    </li>
  );
}

export default TodoItem;
