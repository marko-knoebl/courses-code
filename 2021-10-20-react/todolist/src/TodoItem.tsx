import { Todo } from "./Todo";

import styles from "./TodoItem.module.css";

type TodoItemProps = {
  todo: Todo;
  onChange: (id: number, todo: Partial<Todo>) => void;
  onDelete: (id: number) => void;
};

function TodoItem(props: TodoItemProps) {
  return (
    <li
      className={
        props.todo.completed
          ? `${styles.todo} ${styles.completed}`
          : styles.todo
      }
    >
      <input
        type="checkbox"
        checked={props.todo.completed}
        onChange={(event) => {
          props.onChange(props.todo.id, { completed: event.target.checked });
        }}
      />
      {props.todo.completed ? "DONE:" : "TODO:"} {props.todo.title}
      <button onClick={() => props.onDelete(props.todo.id)}>delete</button>
    </li>
  );
}

export default TodoItem;
