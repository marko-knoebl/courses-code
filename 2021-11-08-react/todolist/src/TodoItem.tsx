import { useContext, useState } from "react";
import { Button } from "react-bootstrap";
import { ThemeContext, useTheme } from "./ThemeContext";
import { Todo } from "./types";

import styles from "./TodoItem.module.css";
import classNames from "classnames";

type TodoItemProps = {
  todo: Todo;
  onDeleteTodo: (id: number) => void;
  onCompletedChange: (id: number, completed: boolean) => void;
  onTitleChange: (id: number, title: string) => void;
};

function TodoItem(props: TodoItemProps) {
  const [titleEditable, setTitleEditable] = useState(false);

  // const themeContext = useContext(ThemeContext);
  const themeContext = useTheme();

  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
    props.onCompletedChange(props.todo.id, event.target.checked);
  }

  let style;
  if (themeContext.theme === "dark") {
    style = { backgroundColor: "#999999" };
  } else {
    style = { backgroundColor: "white" };
  }

  return (
    <li
      style={style}
      className={classNames({
        [styles.todo]: true,
        [styles.completed]: props.todo.completed,
      })}
    >
      <input
        type="checkbox"
        checked={props.todo.completed}
        onChange={handleChange}
      />
      {props.todo.completed ? "DONE" : "TODO"}:{" "}
      {titleEditable ? (
        <input
          value={props.todo.title}
          onChange={(event) =>
            props.onTitleChange(props.todo.id, event.target.value)
          }
        />
      ) : (
        <span>{props.todo.title}</span>
      )}
      <Button onClick={() => props.onDeleteTodo(props.todo.id)}>delete</Button>
      <Button onClick={() => setTitleEditable(!titleEditable)}>
        toggle edit title
      </Button>
    </li>
  );
}

export default TodoItem;
