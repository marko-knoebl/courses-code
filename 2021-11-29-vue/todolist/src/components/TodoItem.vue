<template>
  <li>
    {{ todo.completed ? "DONE: " : "TODO: " }}
    {{ todo.title }}
    <v-btn @click="$emit('toggle', todo.id)" color="secondary">toggle</v-btn>
    <v-btn @click="$emit('delete', todo.id)" color="secondary">delete</v-btn>
  </li>
</template>

<script>
export default {
  name: "TodoItem",
  props: ["todo"],
  // events: ["toggle", "delete"],
};
</script>

<style scoped></style>
