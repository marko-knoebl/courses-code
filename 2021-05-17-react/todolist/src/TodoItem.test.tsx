import TodoItem from "./TodoItem";

import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

test("renders a list item with a given text", () => {
  const title = "test-title";
  render(
    <TodoItem
      title={title}
      completed={false}
      onToggle={() => {}}
      onDelete={() => {}}
    />
  );
  const listItem = screen.getByRole("listitem");
  expect(listItem).toHaveTextContent(new RegExp(title));
});

test("triggers the toggle event", () => {
  const mockFn = jest.fn();
  render(
    <TodoItem
      title="foo"
      completed={false}
      onToggle={mockFn}
      onDelete={() => {}}
    />
  );
  const listItem = screen.getByRole("listitem");
  const toggleBtn = within(listItem).getByRole("checkbox");
  userEvent.click(toggleBtn);
  expect(mockFn).toHaveBeenCalled();
});

test("matches the snapshot", () => {
  render(
    <TodoItem
      title="foo"
      completed={false}
      onToggle={() => {}}
      onDelete={() => {}}
    />
  );
  const li = screen.getByRole("listitem");
  expect(li).toMatchSnapshot();
});
