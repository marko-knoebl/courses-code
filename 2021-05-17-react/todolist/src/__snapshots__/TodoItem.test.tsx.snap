// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<li
  style="text-decoration: none; background-color: lightblue;"
>
  <input
    type="checkbox"
  />
  todo: 
  foo
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      delete
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      toggle
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</li>
`;
